api/src/Filter/ArchivedFilter.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Filter;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\AbstractFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use App\Entity\ArchivableInterface;
+use Doctrine\ORM\QueryBuilder;
+
+final class ArchivedFilter extends AbstractFilter implements GenericFilterInterface
+{
+    private const PARAMETER_NAME = 'archived';
+
+    /**
+     * Filter entities that have a not null "archivedAt" field value.
+     *
+     * @param array<string, mixed> $context
+     */
+    public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, string $operationName = null, array $context = []): void
+    {
+        $archived = $context['filters'][self::PARAMETER_NAME] ?? null;
+        if ('true' !== $archived && $this->supports($resourceClass)) {
+            $queryBuilder->andWhere(sprintf('%s.archivedAt IS NULL', $queryBuilder->getRootAliases()[0] ?? 'o'));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    {
+        try {
+            $reflection = new \ReflectionClass($resourceClass);
+            if (!$reflection->implementsInterface(ArchivableInterface::class)) {
+                throw new \InvalidArgumentException("Can't apply the Archived filter on a resource ({$resourceClass}) not implementing the ArchivableInterface.");
+            }
+        } catch (\ReflectionException $e) {
+            throw new \InvalidArgumentException("Invalid {$resourceClass} resource.");
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return array<string, mixed>
+     */
+    public function getDescription(string $resourceClass): array
+    {
+        $description = 'Generic filter that automatically filters archived entities. One can disable this behavior by passing "true" to this parameter.';
+        $type = 'string';
```

Comment:
There is no need to store it in a variable. Also, shouldn't it be `bool` instead?